kong/clustering/rpc/manager.lua

```diff
@@ -0,0 +1,288 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local server = require("resty.websocket.server")
+local client = require("resty.websocket.client")
+local socket = require("kong.clustering.rpc.socket")
+local concentrator = require("kong.clustering.rpc.concentrator")
+local future = require("kong.clustering.rpc.future")
+local utils = require("kong.clustering.rpc.utils")
+local callbacks = require("kong.clustering.rpc.callbacks")
+local queue = require("kong.clustering.rpc.queue")
+local clustering_tls = require("kong.clustering.tls")
+local constants = require("kong.constants")
+local table_isempty = require("table.isempty")
+local pl_tablex = require("pl.tablex")
+
+
+local ngx_var = ngx.var
+local ngx_ERR = ngx.ERR
+local ngx_log = ngx.log
+local ngx_exit = ngx.exit
+local exiting = ngx.worker.exiting
+local pl_tablex_makeset = pl_tablex.makeset
+local validate_client_cert = clustering_tls.validate_client_cert
+
+
+local WS_OPTS = {
+  timeout = constants.CLUSTERING_TIMEOUT,
+  max_payload_len = kong.configuration.cluster_max_payload,
+}
+local KONG_VERSION = kong.version
+
+
+-- create a new RPC manager, node_id is own node_id
+function _M.new(conf, node_id)
+  local self = {
+    -- node_id: { socket1 => true, socket2 => true, ... }
```

Comment:
should be `-- clients: { socket1 => true, socket2 => true, ... }`?

Reply:
Changed.